### ISSUE TITLE
pose-tensorflow in this project vs eldar's

### ISSUE THREAD

**Issue by cxrodgers** [DeepLabCut/DeepLabCut#78]
Hi
I diffed the pose-tensorflow in this project vs eldar's and there are lots of changes. Mostly there is code that is missing in this version, plus a few changes in defaults and a few bugfixes (e.g. removing references to missing modules, and fully specifying the session name before saving to disk). Other than the bugfixes, I am wondering if there is any reason not to use eldar's version, which is presumably more up to date? For instance, were there any features that were intentionally removed or added?

Also I think it would be better if these modules could be imported so that we don't have to train in the models directory. But there is a lot of namespace confusion in pose-tensorflow, e.g. modules and files that have the same name as variables, and resulting issues with relative imports. I saw in another issue a reference to an ongoing project to "modularize" DeepLabCut. Would it be possible to elaborate on your plans for modularization? I am just wondering if I should create my own fork and do that kind of cleanup, or just wait if there is an impending new version.

Thanks!
Chris

**Comment by AlexEMG** [comment]
Hi Chris,

As stated in [our paper](https://www.nature.com/articles/s41593-018-0209-y) we use Eldar's feature detectors. Those (with some fixes regarding TF versions, defaults etc.) are in DeepLabCut and up-to-date.  Other parts were removed (perhaps not fully, because we internally use some more features).

The **modularized DLC** version will indeed be pip installable, and crucially have all features (extract frames, label frames with pyhon GUIs, create training set, train, evaluate networks as well as analyze videos and much more) as functions. I don't know what your goal with DLC is so, I don't know if you might want to perform your "own cleanup", or wait for the modularized version. However, I hope my description helps you to guide your decision making. 

Cheers,
Alexander

**Comment by cxrodgers** [comment]
Thanks @AlexEMG! I'll work on this in my own fork and if I come up with something useful I will let you know.

**Comment by cxrodgers** [comment]
Hi @AlexEMG just to follow up, I have forked eldar's repository and made it more modular. 
https://github.com/cxrodgers/PoseTF

Everything can now be imported into another script. So for example:

    import PoseTF
    PoseTF.training.train(path_to_config_file)

The data and config file can be located in any arbitrary directory, it doesn't have to be in the 'models' directory. 

I know you're working on your own modularization strategy so you probably have a different plan, but just wanted to mention it in case it is useful.
